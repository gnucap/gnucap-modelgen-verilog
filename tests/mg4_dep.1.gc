
attach ./modelgen_0.so


verilog

`modelgen
module test_dep1(p, n);
	electrical p, n;
	inout p, n;
	real i;
	(*desc="p"*) real p;
	localparam real p1 = 1;
	analog begin
		i = 0;

		if(p1) begin
			i = V(p,n);
		end

		real j;
		j = i;
		p = ddx(j, V(p,n));
		I(p,n) <+ j;
	end
endmodule

!make test_dep1.so > /dev/null
attach ./test_dep1.so

parameter v=1

test_dep1 #() dut(1,0);
vsource #(.dc(v)) v1(1, 0);

list

print dc i(v1) p(dut)

dc
end
