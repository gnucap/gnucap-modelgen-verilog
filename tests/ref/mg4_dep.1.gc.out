parameter v=1 
test_dep1 #() dut (.p(1),.n(0));
vsource #(.dc(v)) v1 (.p(1),.n(0));
#           i(v1)      p(dut)    
 0.        -0.99999    1.        
